tensorflow_datasets/image/spine_web.py

```diff
@@ -0,0 +1,100 @@
+    doi = {10.1007/978-3-319-66182-7_15}
+}
+"""
+
+_DESCRIPTION = """\
+The dataset consists of 609 spinal anterior-posterior x-ray images; it is dataset 16 on SpineWeb.
+The Cobb angles for each image were calculated using landmarks, where four landmarks denoted one vertebrae.
+Here, the training labels are a tensor of 3 cobb angles, corresponding to thoracic, proximal thoracic, and thoracolumbar/lumbar
+"""
+
+
+class SpineWeb(tfds.core.GeneratorBasedBuilder):
+  """SpineWeb"""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=(None, None, 1)),  # b&w jpeg
+            # main thoracic, proximal thoracic, thoracolumbar/lumbar cobb angles
+            "label": tfds.features.Tensor(shape=(3,), dtype=tf.float32)
+        }),
+        supervised_keys=('image', 'label'),
+        urls=['http://spineweb.digitalimaginggroup.ca'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    dl_paths = dl_manager.download_and_extract({
+        'train': 'https://spineweb.s3.amazonaws.com/training_images.zip',
+        'train_csv': 'https://spineweb.s3.amazonaws.com/training_angles.csv',
+        'test': 'https://spineweb.s3.amazonaws.com/test_images.zip',
+        'test_csv': 'https://spineweb.s3.amazonaws.com/test_angles.csv'
+    })
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            # These kwargs will be passed to _generate_examples
+            gen_kwargs={
+                'images_dir_path': dl_paths['train'],
+                'labels': dl_paths['train_csv']
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            # These kwargs will be passed to _generate_examples
+            gen_kwargs={
+                'images_dir_path': dl_paths['test'],
+                'labels': dl_paths['test_csv']
+            },
+        ),
+    ]
+
+  def _generate_examples(self, images_dir_path, labels):
+    """Yields examples."""
+    unordered_list = tf.io.gfile.listdir(images_dir_path)
+    image_names_list = sorted(unordered_list)
+    with tf.io.gfile.GFile(labels, 'r') as f:
+        labels_list = [tf.convert_to_tensor(np.array(line).astype(np.float32)) for line in csv.reader(f)]
+    for image_name, label in zip(image_names_list, labels_list):
+        file_path = "%s/%s" % (images_dir_path, image_name)
+        # if image_name.endswith('.jpg'):
+        #     f = tf.io.gfile.GFile(file_path)
+        #     img = imageio.imread(f, as_gray=True, pilmode='L')
+        #     img = img.astype(np.uint8)
+        #     img = img[..., None]
+        # else:
+        #     img = image_name
+
+        # img = tf.io.read_file(file_path)
+        # img = tf.image.decode_jpeg(img, channels=1).numpy()
```

Comment:
Does using the string method for storing images cause any issues? @chriszhou0916

Reply:
No it works fine. And we confirmed that it produces single channel images.
Using scipy/imageio doesn't work because the file path it receives contains ".zip" and that makes it think it's a zip file. Download_and_extract should've removed the .zip when they extracted the files. 

Leo found that the problem was with using Tensor to store the labels. The tester works fine if we use a feature dictionary instead. The Tensor causes eager/graph mode issues it seems.